proposals/simd-0335-parent-ready-update-marker.md

```diff
@@ -0,0 +1,347 @@
+/// ├─────────────────────────────────────────┤
+/// │ Variant Data              (variable)    │
+/// └─────────────────────────────────────────┘
+///
+/// The byte length field indicates the size of the variant data that follows,
+/// allowing for proper parsing even if unknown variants are encountered.
+pub enum BlockMarkerV2 {
+    BlockFooter(VersionedBlockFooter),
+    ParentReadyUpdate(VersionedParentReadyUpdate),
+}
+```
+
+### Serialization Format
+
+When serialized within a `BlockComponent`, the complete structure is:
+
+```
++---------------------------------------+
+| Entry Count = 0             (8 bytes) |
++---------------------------------------+
+| Marker Version = 2          (2 bytes) |
++---------------------------------------+
+| Variant ID = 1              (1 byte)  |
++---------------------------------------+
+| Length = 41                 (2 bytes) |
++---------------------------------------+
+| Version = 1                 (1 byte)  |
++---------------------------------------+
+| Parent Slot                 (8 bytes) |
++---------------------------------------+
+| Parent Block ID            (32 bytes) |
++---------------------------------------+
+
+Total overhead: 54 bytes per marker
+```
+
+## Alternatives Considered
+
+- **No Explicit Marking / Do Nothing**.
+  Relying on implicit detection of parent switches was considered but rejected
+  because of:
+  - Ambiguity in determining whether a parent switch occurred vs. a duplicate
+    block was received
+  - Inability to validate leader behavior
+  - Lack of auditability for fast leader handover decisions
+
+- **Special Transaction Type**.
+  Using a special transaction to signal parent switches was considered but
+  rejected because:
+  - Higher overhead due to transaction processing requirements
+  - Having a transaction "undo" the state transitions induced by other
+    transactions is ill-advised
+
+- **Extended Block Footer**.
+  Extending the block footer to include parent switch information was
+  considered but is inadequate because:
+  - Parent switches occur during block production, not at the end
+  - The footer appears only at block completion
+  - We cannot accurately represent the switch point within the block
+
+- **Shred Modification**.
+  Modifying the shred structure to include parent switch metadata was
+  considered but rejected due to:
+  - Significant protocol changes required
+  - Backward compatibility challenges
+  - Complexity of implementation
+  - Impact on existing shred processing infrastructure
+
+## Impact
+
+- **Positive**:
+  - Enables full implementation of Alpenglow's fast leader handover
+  - Fast leader handover, upon implementation, reduces block production
+    latency
+  - Minimal space overhead (54 bytes per switch event)
+  - Independent versioning allows evolution without affecting other variants
+
+- **Negative**:
+  - Slight increase in block size when parent switches occur
+
+## Security Considerations
```

Comment:
not sure if we care to call this out since it is more related to the actions we will take upon observing `ParentReadyUpdate`, but one security thing that will require special care is ensuring any transaction cost budgets  are not exceeded for a block even in the presence of a parent update resetting state